hoot-core/src/main/cpp/hoot/core/cmd/DeriveChangesetCmd.cpp

```diff
@@ -295,6 +296,13 @@ class DeriveChangesetCmd : public BaseCommand
     map->visitRw(removeElementsVisitor);
     _currentTaskNum++;
 
+    //  Truncate tags over 255 characters to push into OSM API
+    progress.set(
+      (float)(_currentTaskNum - 1) / (float)_numTotalTasks, "Preparing tags for changeset...");
+    ApiTagTruncateVisitor truncateTags;
+    map->visitRw(truncateTags);
+    _currentTaskNum++;
+
```

Comment:
There are two code paths for datasets, here the map is loaded completely in memory so we can just run the visitor on the map.  The other is below in the filtered stream that adds the visitor to the list of visitors that are called as each element is streamed.